### ISSUE TITLE
Proper Path Format Returns Ruby Compiler Error

### ISSUE THREAD

**Issue by KagemniKarimu** [corck/sia_skynet#1]
upload_file method does not accept basic inputs of '/log.txt', 'log.txt'. or './log.txt'
All seem to return (JSON-related?) error-codes and messages, respectively.
Please advise.

**Comment by corck** [comment]
Can you post the code you use? 

just tried following which worked.
```
touch /tmp/log.txt
...
client = Skynet::Client.new
client.upload_file '/tmp/log.txt'
 => "sia://AABpLbQig1a5kmxdp72vavfthxy4pDJ29fgNWpsypq68BA"
```

**Comment by corck** [comment]
sorry for the late reply, Github notification ended up in Spam

**Comment by KagemniKarimu** [comment]
If I'm not mistaken, your example indicates Linux folder structure. How does it work on Windows File System/Directory?

`a = Skynet::Client.new`
`p a.upload_file ('./TESTfile')`
`p File.open('./TESTfile')`
 
 Produces an error on upload_file,
  ``859: unexpected token at '' (JSON::ParserError)``
  but works fine for File.open
  
  Setting path as either
 `'C:\Users\Admin\RubymineProjects\skynet_test\TESTfile'`
 `'C:/Users/Admin/RubymineProjects/skynet_test/TESTfile'`
 and both produce the same error.

**Comment by corck** [comment]
can you post the full stack trace. I've no windows computer so I can't debug it, but maybe I can figure out the problem.

**Comment by KagemniKarimu** [comment]
Yes. The upload_file example above produces the following backtrace:

```C:/Ruby31-x64/lib/ruby/3.1.0/json/common.rb:216:in `parse': 859: unexpected token at '' (JSON::ParserError)```
        ```from C:/Ruby31-x64/lib/ruby/3.1.0/json/common.rb:216:in `parse'```
       ``` from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/skynet_ruby-0.3.1/lib/skynet/client.rb:220:in `format_response'```
       ``` from C:/Ruby31-x64/lib/ruby/gems/3.1.0/gems/skynet_ruby-0.3.1/lib/skynet/client.rb:79:in `upload_file'```
        ```from C:/Users/Admin/RubymineProjects/skynet_test/skynet.rb:6:in `<main>'```

**Comment by KagemniKarimu** [comment]
Any luck with this? This is by far the most robust ruby gem for Skynet. I would **love** to use it in my project.

**Comment by corck** [comment]
I was checking into it a bit. are you sure you have set the Skynet token properly. Without knowing exactly it looks like the return of skynet might be a "unauthorized message" that can't be parsed. Would you be able to put a `puts res.body` into client.rb at line 220, so before it fails so we see what is returned from Skynet? that would help me.

**Comment by KagemniKarimu** [comment]
Per your request - I tried to puts the _results_:
Injection of `puts res.body` into client.rb at line 220 outputs a blank line.
Injection of `p res.body` into client.rb at line 220 outputs `""`.

I took it a step further and went one step backward in the trace...
Injection of `p res` into client.rb at line 78 outputs `#<Typhoeus::Response:0x00000213ca9d2d00 @options={:httpauth_avail=>0, :total_time=>0.171342, :starttransfer_time=>0.0, :appconnect_time=>0.0, :pretransfer_time=>0.0, :connect_time=>0.113932, :namelookup_time=>0.072426, :redirect_time=>0.0, :effective_url=>"https://siasky.net/skynet/skyfile/?portal=https%3A%2F%2Fsiasky.net&portal_path=%2Fskynet%2Fskyfile&user_agent=Sia-Agent&portal_file_fieldname=file&portal_directory_file_fieldname=files%5B%5D&filename=file&dirname=", :primary_ip=>"135.148.55.62", :response_code=>0, :request_size=>0, :redirect_count=>0, :size_upload=>0.0, :size_download=>0.0, :speed_upload=>0.0, :speed_download=>0.0, :return_code=>:bad_content_encoding, :response_headers=>"", :response_body=>"", :debug_info=>#<Ethon::Easy::DebugInfo:0x00000213caa1b668 @messages=[]>}, @request=#<Typhoeus::Request:0x00000213ca9c4ac0 @base_url="https://siasky.net/skynet/skyfile/", @original_options={:method=>:post, :headers=>{}, :params=>{:portal=>"https://siasky.net", :portal_path=>"/skynet/skyfile", :user_agent=>"Sia-Agent", :portal_file_fieldname=>"file", :portal_directory_file_fieldname=>"files[]", :filename=>"file", :dirname=>""}, :body=>{:file=>#<File:C:\Users\Admin\RubymineProjects\gemtest\Gemfile (closed)>}}, @options={:method=>:post, :headers=>{"User-Agent"=>"Sia-Agent", "Expect"=>""}, :params=>{:portal=>"https://siasky.net", :portal_path=>"/skynet/skyfile", :user_agent=>"Sia-Agent", :portal_file_fieldname=>"file", :portal_directory_file_fieldname=>"files[]", :filename=>"file", :dirname=>""}, :body=>{:file=>#<File:C:\Users\Admin\RubymineProjects\gemtest\Gemfile (closed)>}, :maxredirs=>50}, @on_progress=[], @response=#<Typhoeus::Response:0x00000213ca9d2d00 ...>, @on_complete=[], @on_failure=[]>>
`

**Comment by corck** [comment]
ok, thx, bad conent encoding sounds like the problem.

**Comment by KagemniKarimu** [comment]
Got it! Thoughts on solution?

**Comment by corck** [comment]
... started to look into quickly but failed to get Ruby running on my Windows VM. Was complaining about a missing dll related to curl. things I tried failed so couldn't work on it. Need to find time to get that running.

**Comment by corck** [comment]
... I can't get the curl dll set up on my VM. tried every post on Stack Overflow. How did you get that running on your machine?

**Comment by KagemniKarimu** [comment]
That's the issue #2 mentioned with this gem. 

This is essentially the steps I took to remedy the issue:
https://stackoverflow.com/a/54520917

**Comment by KagemniKarimu** [comment]
No joy?

**Comment by corck** [comment]
got it running but couldn't figure out yet, why its not working. response information doesn't include any valuable information. will need to dig more deeper into it.
